telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt

```diff
@@ -40,21 +44,32 @@ object TelemetryGenerator {
                     .addParameter("name", String::class)
                     .build()
             )
+            .addFunction(FunSpec.builder("toString").addModifiers(KModifier.OVERRIDE).returns(String::class).addStatement("return name").build())
+            .addKdoc(item.description)
+
         item.allowedValues!!.forEach { enumValue ->
             enum.addEnumConstant(
                 enumValue.toString().toUpperCase().filterInvalidCharacters(), TypeSpec.anonymousClassBuilder()
                     .addSuperclassConstructorParameter("%S", enumValue.toString())
                     .build()
             )
         }
-        enum.addFunction(FunSpec.builder("toString").addModifiers(KModifier.OVERRIDE).returns(String::class).addStatement("return name").build())
-        enum.addFunction(
-            FunSpec.builder("from")
-                .returns(ClassName("", item.name.toTypeFormat()).copy(nullable = true))
-                .addParameter("type", Any::class)
-                .addStatement("return values().firstOrNull { it.name == type.toString() }").build()
-        )
-        enum.addKdoc(item.description)
+
+        val companion = TypeSpec.companionObjectBuilder()
+            .addFunction(
+                FunSpec.builder("from")
+                    .returns(ClassName("", item.name.toTypeFormat()).copy(nullable = true))
+                    .addParameter("type", Any::class.asTypeName().copy(nullable = true))
```

Comment:
Similarly, why can't this be non-nullable?

Reply:
back to being non nullable